### DISCUSSION TITLE
Two Vue 2 Monitors in one Panel sharing data?

### DISCUSSION THREAD

**Discussion by Shesakillatwo** [emporia-vue-local/esphome#271] [Q&A]
I have two esphome flashed Emporia 2 devices in one Breaker panel to monitor about 30 circuits.  One of these has the Main CT clamps installed and the other does not.  A few quick questions.

1. SImple question is, is there a way to make all the 30 circuits to look and act like on BIG Vue 2 before the data gets to Home Assistant so Totals and Balance are accurate for the entire panel? If not,.......
2. Are the main A & B Voltages, Frequency and Phase Angle measured on the Main CT Clamps or on the Red and Black Power wires?
3. Since each Vue 2 has about half of the total breakers in the panel the Balance for each does not actually reflect the Balancce of unmeasured circuits in the panel.  Can I combine all the individual circuits from both and then determine what the balance of the panel power is after they are combined?  Can I do this before I get the data into Home Assistant?
4. For the Vue 2 without Main CT Clamps, can I also easily measure the Sum of all the individual measured circuits and add this to the balance - total power of the Vue 2 with the CT Clamps?

**Comment by oliv3r** [comment]
> I have two esphome flashed Emporia 2 devices in one Breaker panel to monitor about 30 circuits. One of these has the Main CT clamps installed and the other does not. A few quick questions.

I am **planning** also to install two, to also monitor about 30 circuits. This in the EU with 230V 3 Phases. But do share your experiences so far!!

> 
>     1. SImple question is, is there a way to make all the 30 circuits to look and act like on BIG Vue 2 before the data gets to Home Assistant so Totals and Balance are accurate for the entire panel? If not,.......

I have no idea, but I expect no. Simple reason is that each vue would have its own esphome connecting to HA. To do what you want you'd have to get the esp's talking to eachother first (bluetooth, wifi, direct cable link). You could remove the esp from one of the boards, and connect the afaik i2c bus from one vue to the other, give it a different address (or a different i2c bus). That would require some soldering however.

Off-the-shelf, probably not. You could still do smart things within HA of course, which are improvements that seem quite sensible to me regardless.

> 
>     2. Are the main A & B Voltages, Frequency and Phase Angle measured on the Main CT Clamps or on the Red and Black Power wires?

You should be able to get this from the schematics that we RE-ed. https://github.com/emporia-vue-local/emporia-vue2-reversing/tree/master/kicad not perfect, but good enough to answer your question for sure. But in short, the Voltage, phase angle and frequency is measured on the molex-like connector, the current on the big CT coils.

Afaik the 3 phases (2 on split-phase) are measured using the atmega adc, the 16 circuits use the two analog switches and a single (or two, don't remember) adc's.

Though i think the atmega even has only 1 single adc with a multiplex. Better to have picked a dual ADC mcu ;)

> 
>     3. Since each Vue 2 has about half of the total breakers in the panel the Balance for each does not actually reflect the Balance of unmeasured circuits in the panel.  Can I combine all the individual circuits from both and then determine what the balance of the panel power is after they are combined?  Can I do this before I get the data into Home Assistant?

No idea what you are asking about with regards to balance, but point two is probably the same as point 1 above.

> 
>     4. For the Vue 2 without Main CT Clamps, can I also easily measure the Sum of all the individual measured circuits and add this to the balance - total power of the Vue 2 with the CT Clamps?

I think within HA you can do whatever you want. AFAIK you can even use the main CT clamps as 'regular CT clamps' in terms of energy usage. Nothing 'forces' you to use the main clamps only for the main connections, just the fact that the CT clamps themselves are bigger and can handle higher currents.

**Reply by FedgarLurch** [reply]
I am doing this right now. I have 28 circuits, 120 / 240 V panel. One of the Vues has the main CT's and it is called Primary.
The other one is called "Secondary". What my plan is, is to make HA be the broker for the signals, some kind of templating, and then have the HA look at that instead of the direct sensor readings. 

Rather than messing with the hardware, just have everything report to HA, have it combine / multiply / divide / etc and then have HA look at those values. Similar to how in the code we have copy values. This would effectively be another layer of that. 

Anyone else done anything like this?

**Reply by clowrey** [reply]
you could also do it in ESPhome with local communication https://esphome.io/cookbook/http_request_sensor.html

**Reply by FedgarLurch** [reply]
Good call. I will look at that this weekend. I have never had to do that, would it be a master slave thing like modbus or how does that work? 
Nevermind, ill read the docs. (Sorry, its so easy to ask people) lol

**Reply by clowrey** [reply]
it may not be the best way as they say not to overload the server with too many requests.. just an idea.. you can definitely create composite sensors in HA too, I haven't really had to do that. But do have 3x emporias at one site, but all different panels and loads etc

**Reply by FedgarLurch** [reply]
I agree, I don't like having tons of extra unnecessary devices in HA. 
Could you share your code (or whatever portions)? Just so we have a different perspective on how to do that? 

Thanks!

**Comment by gekkehenkie11** [comment]
Anybody any luck on this? I Actually have 3 VUE's that I'd like to combine into 1

**Reply by Shesakillatwo** [reply]
I found no way to do it in ESP Home which is what I wanted to do.  Instead took a combined path by doing two things.  I modified some of the YAML files for some of the Vue Devices and I also created some entities in Home Assistant to add together different values received from the various Vue Devices to get to some of the data points I wanted.

**Comment by gekkehenkie11** [comment]
Actually I see that the original firmware now has an option to group multiple VUE's into 1 big panel. Would be a shame if we can't do the same in Home assistant! :)

**Reply by Shesakillatwo** [reply]
There may be a way, but I could not figure it out.   If you do please drop a note here!  As a totally side note.  After several months these work great with ESPHome!

**Comment by oliv3r** [comment]
I think virtually in HA there surely must be a way, or a feature request is needed. One virtual vue ;) I know you can hide devices I. HA, while still having them there ...

**Comment by ThomasHFWright** [comment]
Did anyone get this working in a sensible way with more than one vue in a panel?

**Reply by gekkehenkie11** [reply]
yeah. You can just create a dashboard in home assistant. Then add the "entities card" to it. And then you can just select all sensors from all of your VUE's into one card. Or, what I did, you can create a dashboard with multiple entities cards and then group them like you want. For example I have 4 VUE's in my house and I think 3 of them have Air conditioners. I made a card called 'air conditioners' and there I now have all AC's from all the VUE's grouped into that. So there's a LOT you can do. Even more so if you add the Lovelace card-mod you can do all kinds of crazy stuff. But something simple I do using that card is that I show any sensor that's above 100W, in color red. So I can see quickly what's consuming most of my energy.

Regarding the totals: You should only have 1 VUE monitor your mains. And then I calculate the total per VUE by adding each entry. Then I add those VUE totals up and that should be the same as the main monitor. In my case it's usually within 3%.
Alternatively you can create a 'helper' to add up all 50A sensors and use that to compare to your mains. You can really do pretty much anything in HA.
